src/libraries/System.IO.Compression/ref/System.IO.Compression.cs

```diff
@@ -110,6 +110,7 @@ public partial class ZipArchiveEntry
         public System.IO.Compression.ZipArchive Archive { get { throw null; } }
         [System.Diagnostics.CodeAnalysis.AllowNull]
         public string Comment { get { throw null; } set { } }
+        public bool IsEncrypted { get { throw null; } }
```

Comment:
Is there anywhere we should be consuming this, like the various `ZipFile.Extract*` methods?

Reply:
I considered that in the API proposal in the [Additional APIs to Consider section](https://github.com/dotnet/runtime/issues/68897#user-content-zipfile-accelerators). Any change to the existing `ZipFile.Extract*` methods' behavior could be breaking though, so I held off on changing any of the existing methods.  I got confirmation from a partner team that this simple boolean will substantially improve their scenarios, so I figured it best to hold off on other changes/additions until we implement [Add password to ZipArchive · Issue #1545](https://github.com/dotnet/runtime/issues/1545).

With that said, if you have specific recommendations here it wouldn't necessarily be breaking, I'm open to ideas.

Reply:
> Any change to the existing ZipFile.Extract* methods' behavior could be breaking though

Agreed, it would be breaking. I think we should open a new issue to discuss if we want to modify the logic of these extraction methods so that encrypted entries get skipped (since we don't yet have an API to provide a password).

Here's a summary of where the changes would be done:

The `ZipFile.Extract*` and `ZipFileExtensions.Extract*` methods all eventually [call this](https://source.dot.net/#System.IO.Compression.ZipFile/System/IO/Compression/ZipFileExtensions.ZipArchiveEntry.Extract.cs,87):

```cs
ExtractRelativeToDirectory(this ZipArchiveEntry source, string destinationDirectoryName, bool overwrite)
```

The method then does some validations, then [extracts the entry here](https://source.dot.net/#System.IO.Compression.ZipFile/System/IO/Compression/ZipFileExtensions.ZipArchiveEntry.Extract.cs,117), which is where we would check for encryption:

```cs
source.ExtractToFile(fileDestinationPath, overwrite: overwrite);
```